bus-mapping/src/eth_types.rs

```diff
@@ -137,8 +137,9 @@ impl ToWord for Address {
 impl<F: FieldExt> ToScalar<F> for Address {
     fn to_scalar(&self) -> Option<F> {
         let mut bytes = [0u8; 32];
-        bytes[32 - Self::len_bytes()..].copy_from_slice(self.as_bytes());
-        F::from_bytes(&bytes).into()
+        bytes[..Self::len_bytes()].copy_from_slice(self.as_bytes());
+        let bytes_f = F::from_bytes(&bytes);
```

Comment:
That's correct! `reverse` should do. Although you should be able to transform `Address` to `Word`. And with `Word` choose the endianness directly instead of reversing by yourself.

Reply:
cool, I would prefer `reverse` does here :)

Reply:
fixed in 7ad3636